src/main/java/com/heroku/deployer/Main.java

```diff
@@ -159,13 +161,14 @@ public Integer call() throws Exception {
         }
 
         Path sourceBlobArchive = SourceBlobPackager.pack(sourceBlobDescriptor);
+        String sourceBlobArchiveSha256 = DigestUtils.sha256Hex(Files.newInputStream(sourceBlobArchive));
 
         DeploymentDescriptor deploymentDescriptor = new DeploymentDescriptor(
                 appName.get(),
                 buildpacks,
                 Collections.emptyMap(),
                 sourceBlobArchive,
-                GitUtils.getHeadCommitHash(projectDirectory).orElse("unknown"));
+                GitUtils.getHeadCommitHash(projectDirectory).orElse("nogit-" + sourceBlobArchiveSha256));
```

Comment:
I see the old plugins used `unknown` too, so I presume this isn't a regression specific to `heroku-jvm-application-deployer`, but a fix to an ~existing issue? eg:
https://github.com/heroku/heroku-jvm-application-deployer/blob/c6137ca346707cd0453c9a49d9b8b8b9db1f3fba/src/main/java/com/heroku/deployer/Main.java#L168

Though I see the support ticket was about `jar:deploy` which uses https://github.com/heroku/plugin-java , which uses a `heroku-deploy-complete` jar file, whose source repo I can't even find. So I guess maybe `jar:deploy` doesn't use those plugins anyway?

Reply:
Oh I thought I was searching `heroku-maven-plugin` for the "unknown" string, but that repo 301 redirects to `heroku-jvm-application-deployer` so it turns out the search was against this repo after all. So that search result wasn't from an old plugin.

Everything to do with the Java plugins is super confusing to me. Glad we've almost at the end of being able to sunset the old things to simplify the story here :-)